util/syn_yosys.sh

```diff
@@ -10,59 +10,139 @@
 # The following tools are required:
 #  - sv2v: SystemVerilog-to-Verilog converter from github.com/zachjs/sv2v
 #  - yosys: synthesis tool from github.com/YosysHQ/yosys
+#  - Cadence Conformal
 #
 # Usage:
-#   syn_yosys.sh 2>&1 | tee syn.std
+#   ./syn_yosys.sh |& tee syn.std
 
 #-------------------------------------------------------------------------
 # use fusesoc to generate files and file list
 #-------------------------------------------------------------------------
-\rm -Rf build
-fusesoc --cores-root .. run --target=sim --setup --build formal > /dev/null 2>&1
+rm -Rf build syn_out
+
+fusesoc --cores-root .. run --tool=icarus --target=syn \
+  --setup "lowrisc:systems:top_earlgrey" > /dev/null 2>&1
 
 # copy all files into directory "syn_out"
-\rm -Rf syn_out
 mkdir syn_out
-cp build/formal_0/src/*/*/*.sv build/formal_0/src/*/*/*/*.sv syn_out
+cp build/*/src/*/*.sv* build/*/src/*/*/*.sv* build/*/src/*/*/*/*.sv* syn_out
+cd syn_out
+
+# copy file list and remove incdir lines
+grep -v incdir ../build/*/*/*.scr > flist_gold
+
+# remove all hierarchical paths
+sed -i 's!.*/!!' flist_gold
+
+# generate revised flist by replacing '.sv' by '.v' and removing packages
+sed 's/.sv/.v/g' flist_gold | grep -v "_pkg.v" > flist_rev
 
 #-------------------------------------------------------------------------
 # convert all RTL files to Verilog
 #-------------------------------------------------------------------------
-cd syn_out
 
+# hack SystemVerilog files to avoid SV2V and/or LEC issues
+# TODO: eventually remove these hacks
+
+# 1) Setting ByteAccess = 0 for tlul_adapter_sram.sv doesn't work. This
+# hack changes the functionality so needs fixing
+sed -i 's/\.ByteAccess(0)/\.ByteAccess(1)/g' *.sv
+sed -i 's/\.ByteAccess  (0)/\.ByteAccess(1)/g' *.sv
+
+# 2) Replace "inside" experessions
+sed -i 's/state_q inside {Read, WaitReadValid}/((state_q == Read) || (state_q == WaitReadValid))/' dmi_jtag.sv
+sed -i 's/k inside {TckIdx, TmsIdx, TrstIdx, SrstIdx, TdiIdx, TdoIdx}/(k==TckIdx)||(k==TmsIdx)||(k==TrstIdx)||(k==SrstIdx)||(k==TdiIdx)||(k==TdoIdx)/' jtag_mux.sv
+sed -i 's/(csr_op_i inside {CSR_OP_WRITE,/((csr_op_i==CSR_OP_WRITE)||(csr_op_i==CSR_OP_SET)||(csr_op_i==CSR_OP_CLEAR));/' ibex_cs_registers.sv
+sed -i 's/    CSR_OP_SET,//' ibex_cs_registers.sv
+sed -i 's/    CSR_OP_CLEAR});//' ibex_cs_registers.sv
+sed -i "s/logic'(tl_i\.a_opcode inside {PutFullData, PutPartialData})/((tl_i\.a_opcode == PutFullData) || (tl_i\.a_opcode == PutFullData))/" tlul_adapter_sram.sv
```

Comment:
This all looks very magical to me. I think that at least the `CSR_OP_*` lines need some comments explaining what they do.

Reply:
Yes, these are temp hacks, some of them might not be needed anymore. Will address it in the new PR